mmseg/core/optimizers/layer_decay_optimizer_constructor.py

```diff
@@ -110,14 +135,22 @@ def add_params(self, params, module):
             else:
                 group_name = 'decay'
                 this_weight_decay = weight_decay
-
-            if decay_type == 'layer_wise':
-                layer_id = get_num_layer_layer_wise(
-                    name, self.paramwise_cfg.get('num_layers'))
-                logger.info(f'set param {name} as id {layer_id}')
+            if 'layer_wise' in decay_type:
+                if 'ConvNeXt' in module.__class__.__name__:
+                    layer_id = get_num_layer_layer_wise(
+                        name, self.paramwise_cfg.get('num_layers'))
+                    logger.info(f'set param {name} as id {layer_id}')
+                elif 'BEiT' in module.__class__.__name__:
+                    layer_id = get_num_layer_for_vit(name, num_layers)
+                    logger.info(f'set param {name} as id {layer_id}')
+                else:
+                    raise NotImplementedError()
             elif decay_type == 'stage_wise':
-                layer_id = get_num_layer_stage_wise(name, num_layers)
-                logger.info(f'set param {name} as id {layer_id}')
+                if 'ConvNeXt' in module.__class__.__name__:
+                    layer_id = get_num_layer_stage_wise(name, num_layers)
```

Comment:
it might be better to change `get_num_layer_stage_wise` to `get_num_stage_for_convnext`.